### PR TITLE
print friendly error message if btrfs tells us /dev/root was backing block device of root fs

### PR DESCRIPTION
**Pull request by poettering**
Prompted by #16953

### CODE REVIEW

File: src/gpt-auto-generator/gpt-auto-generator.c [hunk 1]
```diff
@@ -759,6 +768,13 @@ static int add_mounts(void) {
         }
 
         return enumerate_partitions(devno);
+
+euclean:
+        return log_error_errno(
+                        r,
+                        "File system behind %s system is reported by btrfs to be backed by pseudo-device /dev/root, which is not a valid user-space accessible device node. "
```
Comment (keszybz) [inline src/gpt-auto-generator/gpt-auto-generator.c]: "system" seems misplaced here, we'll have "File system behind root file system system".
user-space → userspace for consistency with the rest of the codebase.

File: src/gpt-auto-generator/gpt-auto-generator.c [hunk 2]
```diff
@@ -725,14 +725,23 @@ static int enumerate_partitions(dev_t devnum) {
...
 static int add_mounts(void) {
+        const char *euclean_path;
         dev_t devno;
         int r;
 
         r = get_block_device_harder("/", &devno);
+        if (r == -EUCLEAN) {
+                euclean_path = "root file system";
+                goto euclean;
+        }
```
Comment (keszybz) [inline src/gpt-auto-generator/gpt-auto-generator.c]: I think it'd be nicer to add an inline helper function that takes this label as an argument and wraps `log_error_errno()` with the appropriate message. This way the message itself will not have to be defined in a thousand places and goto will not need to be used.

**Review by keszybz — CHANGES_REQUESTED** [review]

File: src/gpt-auto-generator/gpt-auto-generator.c (hunk 2 above)
Comment (poettering) [reply src/gpt-auto-generator/gpt-auto-generator.c]: makes sense

**Comment by poettering** [comment]
Force pushed a new version. PTAL. Only the suggested changes.

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**